ios/ImagePickerManager.m

```diff
@@ -292,10 +314,114 @@ + (UIImage*)getUIImageFromInfo:(NSDictionary*)info {
 + (NSURL*)getNSURLFromInfo:(NSDictionary*)info {
     if (@available(iOS 11.0, *)) {
         return info[UIImagePickerControllerImageURL];
-    } else {
+    }
+    else {
         return info[UIImagePickerControllerReferenceURL];
     }
 }
 
 @end
 
+@implementation ImagePickerManager(PHPickerViewControllerDelegate)
+
+- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14))
+{
+    if (results.count == 0) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [picker dismissViewControllerAnimated:YES completion:^{
+                self.callback(@[@{@"didCancel": @YES}]);
+            }];
+        });
+        return;
+    }
+
+    for (PHPickerResult *result in results) {
+        NSItemProvider *provider = result.itemProvider;
+
+        if ([provider canLoadObjectOfClass:[UIImage class]]) {
+            [provider loadObjectOfClass:[UIImage class]
+                      completionHandler:^(__kindof id<NSItemProviderReading> _Nullable object, NSError * _Nullable error) {
+                UIImage *image = object;
+                NSDictionary *response = [self makeResponseFromImage:image];
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [picker dismissViewControllerAnimated:YES completion:^{
+                        self.callback(@[response]);
+                    }];
+                });
+            }];
+        }
+        else if ([provider hasItemConformingToTypeIdentifier:@"public.movie"]) {
+            [provider loadFileRepresentationForTypeIdentifier:@"public.movie"
+                                            completionHandler:^(NSURL * _Nullable url, NSError * _Nullable error) {
+                NSDictionary *response = [self makeResponseFromURL:url];
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [picker dismissViewControllerAnimated:YES completion:^{
+                        self.callback(@[response]);
+                    }];
+                });
+            }];
+        }
+    }
+}
+
+- (NSDictionary *)makeResponseFromImage:(UIImage *)image {
```

Comment:
makeResponseFromImage and onImageObtained should be merged into a common function, both perform the same thing but only with different inputs.